exporters/prometheus/exporter.go

```diff
@@ -344,3 +376,27 @@ func sanitizeName(n string) string {
 
 	return b.String()
 }
+
+func validateMetrics(name, description string, metricType *dto.MetricType, mfs map[string]*dto.MetricFamily) (drop bool, help string) {
+	emf, exist := mfs[name]
+	if !exist {
+		mfs[name] = &dto.MetricFamily{
+			Name: proto.String(name),
+			Help: proto.String(description),
+			Type: metricType,
+		}
+		return false, ""
+	}
+	if emf.GetHelp() != description {
+		if emf.GetType() != *metricType && (emf.GetType() == dto.MetricType_HISTOGRAM || *metricType == dto.MetricType_HISTOGRAM) {
+			return false, ""
```

Comment:
It seems like this could cause us to keep a metric with different types (drop=false, when the types are not equal), which I don't believe is correct.

Suggestion: Check `if emf.GetType() != *metricType {` **before** checking the description so that we always drop metrics with different type comments.